llvm/lib/Transforms/Instrumentation/MemProfiler.cpp

```diff
@@ -824,11 +824,22 @@ memprof::extractCallsFromIR(Module &M) {
           continue;
 
         StringRef CalleeName = CalledFunction->getName();
-        uint64_t CallerGUID =
-            IndexedMemProfRecord::getGUID(DIL->getSubprogramLinkageName());
-        uint64_t CalleeGUID = IndexedMemProfRecord::getGUID(CalleeName);
-        LineLocation Loc = {GetOffset(DIL), DIL->getColumn()};
-        Calls[CallerGUID].emplace_back(Loc, CalleeGUID);
+        if (DIL->getInlinedAt()) {
+          for (; DIL; DIL = DIL->getInlinedAt()) {
```

Comment:
Can we simply do this loop in both cases (with and without a getInlinedAt())? In the case of no getInlinedAt() afaict this would execute once and do the same thing as the else case further below.

Reply:
Thank you for pointing this out!  Addressed in the latest revision.